src/protobuf/message.cr

```diff
@@ -3,10 +3,19 @@ module Protobuf
     macro contract
       FIELDS = {} of Int32 => HashLiteral(Symbol, ASTNode)
       {{yield}}
-      _generate_decoder
-      _generate_encoder
+      _generate_decoder "proto2"
+      _generate_encoder "proto2"
```

Comment:
Sounds like this whole macro could just call the `contract_of` macro? Instead of duplicating code.

Or there could only be one macro with an optional parameter for the version and a default value of "proto2"? (I think that's possible)